### PR TITLE
Spark 3.3: Use regular planning for applicable row-level operations

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR switches all merge-on-read and MERGE copy-on-write operations to use built-in Spark planning. This enables more optimizations for applicable row-level commands like discarding pushed down predicates.

### CODE REVIEW

File: spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala [hunk 1]
```diff
@@ -38,8 +40,15 @@ object RowLevelCommandScanRelationPushDown extends Rule[LogicalPlan] with Predic
   import ExtendedDataSourceV2Implicits._
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan transformDown {
+    // use regular planning for delta-based plans and copy-on-write MERGE operations
```
[the remark this answers is not in the document]
Comment (aokolnychyi) [reply spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala:43]: Yep, I can add a more detailed comment.

[the remark this answers is not in the document]
Comment (aokolnychyi) [reply spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala]: Added.

File: spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala [hunk 2]
```diff
@@ -38,8 +40,16 @@ object RowLevelCommandScanRelationPushDown extends Rule[LogicalPlan] with Predic
   import ExtendedDataSourceV2Implicits._
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan transformDown {
+    // use native Spark planning for delta-based plans and copy-on-write MERGE operations
+    // unlike other commands, these plans have filters that can be pushed down directly
```
Comment (RussellSpitzer) [inline spark/v3.3/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/RowLevelCommandScanRelationPushDown.scala:44]: Thanks this makes it much clearer

**Comment by aokolnychyi** [comment]
Thanks for reviewing, @RussellSpitzer!

**Comment by aokolnychyi** [comment]
I'll have to revisit the approach in this PR. It is actually not safe and may lead to data correctness bugs.
I'll follow up with a fix next week.

**Pull request merged**